opm/simulators/flow/RFTContainer.cpp

```diff
@@ -0,0 +1,225 @@
+        }
+
+        //add data infrastructure for shut wells
+        if (!wellDatas.count(well.name())) {
+            auto& wellData = wellDatas[well.name()];
+            wellData.connections.reserve(well.getConnections().size());
+            std::transform(well.getConnections().begin(),
+                           well.getConnections().end(),
+                           std::back_inserter(wellData.connections),
+                           [](const auto& connection)
+                           {
+                               data::Connection res;
+                               res.index = connection.global_index();
+                               return res;
+                           });
+            wellDatas.emplace(std::make_pair(well.name(), wellData));
+        }
+
+        data::Well& wellData = wellDatas.at(well.name());
+
+        auto cond_assign = [](double& dest, unsigned idx, const auto& map)
+        {
+            auto it = map.find(idx);
+            if (it != map.end()) {
+                dest = it->second;
+            }
+        };
+
+        std::for_each(wellData.connections.begin(), wellData.connections.end(),
+                      [&cond_assign, this](auto& connectionData)
+                      {
+                          const auto index = connectionData.index;
+                          cond_assign(connectionData.cell_pressure, index,
+                                      oilConnectionPressures_);
+                          cond_assign(connectionData.cell_saturation_water, index,
+                                      waterConnectionSaturations_);
+                          cond_assign(connectionData.cell_saturation_gas, index,
+                                      gasConnectionSaturations_);
+                       });
+    }
+
+    oilConnectionPressures_.clear();
+    waterConnectionSaturations_.clear();
+    gasConnectionSaturations_.clear();
+}
+
+template<class FluidSystem>
+void RFTContainer<FluidSystem>::
+allocate(const std::size_t reportStepNum)
+{
+    // Well RFT data
+    const auto& rft_config = schedule_[reportStepNum].rft_config();
+    for (const auto& well: schedule_.getWells(reportStepNum)) {
+
+        // don't bother with wells not on this process
+        if (!wellQuery_(well.name())) {
+            continue;
+        }
+
+        if (!rft_config.active()) {
+            continue;
+        }
+
+        for (const auto& connection: well.getConnections()) {
+            const std::size_t index = connection.global_index();
+
+            if (FluidSystem::phaseIsActive(oilPhaseIdx)) {
+                oilConnectionPressures_.emplace(std::make_pair(index, 0.0));
+            }
+
+            if (FluidSystem::phaseIsActive(waterPhaseIdx)) {
+                waterConnectionSaturations_.emplace(std::make_pair(index, 0.0));
+            }
+
+            if (FluidSystem::phaseIsActive(gasPhaseIdx)) {
+                gasConnectionSaturations_.emplace(std::make_pair(index, 0.0));
+            }
```

Comment:
`Emplace(make_pair(i, 0))` is a bit wasteful when you can just do `emplace(i, 0)` instead.

Reply:
yup.